src/services/executionEnvironment.ts

```diff
@@ -0,0 +1,316 @@
+        }
+      })
+    }
+
+    private isPluginInPath(containerName: string, searchPath: string, pluginFolderPath: string): boolean {
+      //const command = this.wrapContainerArgs(`find ${searchPath} -path '${pluginFolderPath}'`)
+      const command = `${this._container_engine} exec -i ${containerName} find ${searchPath} -path '${pluginFolderPath}'`
+      try {
+        this.connection.console.info(`Executing command ${command}`)
+        const result = child_process.execSync(command, {
+          encoding: 'utf-8',
+        }).trim();
+        return result !== ''
+
+      } catch (error) {
+        this.connection.console.error(error)
+        return false
+      }
+    }
+
+    private runContainer(containerName: string): boolean {
+      // ensure container is not running
+      this.cleanUpContainer(containerName)
+      try {
+        const command = `${this._container_engine} run --rm -it -d --name ${containerName} ${this._container_image} bash`
+        this.connection.console.log(`run container with command '${command}'`)
+        child_process.execSync(command, {
+          encoding: 'utf-8'
+        });
+      } catch (error) {
+        this.connection.console.error(`Failed to initialize execution environment '${this._container_image}': ${error}`)
+        return false
+      }
+      return true
+    }
+
+    private async copyPluginDocFiles(
+      hostPluginDocCacheBasePath: string,
+      containerName: string,
+      contianerPluginPaths: string[],
+      searchKind: string
+      ): Promise<string[]> {
+
+      const updatedHostDocPath: string[] = []
+      if (fs.existsSync(hostPluginDocCacheBasePath)) {
+        contianerPluginPaths.forEach((srcPath) => {
+          updatedHostDocPath.push(path.join(hostPluginDocCacheBasePath, srcPath))
+        })
+      } else {
+        contianerPluginPaths.forEach((srcPath) => {
+          if (srcPath === '' || !this.isPluginInPath(containerName, srcPath, searchKind)) { return }
+          const destPath = path.join(hostPluginDocCacheBasePath, srcPath)
+          const destPathFolder = destPath.split(path.sep).slice(0, -1).join(path.sep)
+          fs.mkdirSync(destPath, { recursive: true });
+          const copyCommand = `docker cp ${containerName}:${srcPath} ${destPathFolder}`
+          this.connection.console.log(`Copying plugins from container to local cache path ${copyCommand}`)
+          asyncExec(copyCommand, {
+            encoding: 'utf-8'
+          });
+
+          updatedHostDocPath.push(destPath)
+        })
+      }
+
+      return updatedHostDocPath
+    }
+
+    private updateCachePaths(pluginPaths:string[], cacheBasePath:string): string[] {
+      const localCachePaths: string[] = []
+      pluginPaths.forEach((srcPath) => {
+        const destPath = path.join(cacheBasePath, srcPath)
+        if (fs.existsSync(destPath)) {
+          localCachePaths.push(destPath)
+        }
```

Comment:
The value of `ansibleConfig.collections_paths` is passed to this method for `pluginPaths` and it contains of default configured plugin paths with EE which might not actually exist and since local cache path is a combination of local cache per EE unique path and path within EE (grepped by executing `ansible-config` command within EE) some of the paths might now exist within local cache hence the check.